<html>
<head>
  <title>RNA-seq Data Analysis</title>
  <basefont face="Segoe UI" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="Evernote Windows/308919 (en-US, DDL); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: Segoe UI;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="443"/>
<h1>RNA-seq Data Analysis</h1>

<div>
<span><div><h1>RNA-seq processing</h1><div>The extraction of RNA should begin with the highest quality RNA. An agarose gel or bioanalyzer can be used to assess the quality of the RNA.</div><div><br/></div><div><br/></div><div><img src="RNA-seq Data Analysis_files/Image.jpg" type="image/jpeg" data-filename="Image.jpg"/><img src="RNA-seq Data Analysis_files/Image.png" type="image/png" data-filename="Image.png"/><img src="RNA-seq Data Analysis_files/Image [1].png" type="image/png" data-filename="Image.png"/></div><div><br/></div><div><br/></div><div><br/></div><div><br/></div><div><br/></div><div><img src="RNA-seq Data Analysis_files/Untitled Diagram.jpg" type="image/jpeg" data-filename="Untitled Diagram.jpg"/></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">Request for resources to run the project on the HPCC.</span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">In a tmux pane:</span></div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>srun --x11 --partition=intel -c 4 --mem=8gb  --time 48:00:00 --pty bash -l</div></div><div style="margin-bottom: 14.15pt;"><br/></div><div style="margin-bottom: 14.15pt;"><font style="font-size: 14pt;"><span style="font-size: 14pt; font-weight: bold;">I. Trimming adapters</span></font></div><div style="margin-bottom: 14.15pt;"><span style="font-size: 10pt;">Libraries prepared for the illumina platform (NEB kits) must be trimmed for adapters only at the 3' end. Adapters are not found on the 5' ends.</span></div><div style="margin-bottom: 14.15pt;"><img src="RNA-seq Data Analysis_files/Image [2].png" type="image/png" data-filename="Image.png"/></div><div style="margin-bottom: 14.15pt;"><img src="RNA-seq Data Analysis_files/Image [3].png" type="image/png" data-filename="Image.png"/></div><div><span style="text-decoration: underline;">5'- 5' universal Adapter (58 bp)</span></div><div><span style="color: rgb(186, 0, 255);">     P5 primer (binds to flow cell)         |</span>------------<span style="color: rgb(54, 101, 238);">-sequencing primer-</span>---------------|</div><div><span style="color: rgb(186, 0, 255);">5'--AATGATACGGCGACCACCGA</span>GATCT<span style="color: rgb(54, 101, 238);">ACACTCTTTCCCTACACGACGCTCTTCCGATCT</span></div><div>                                              </div><div><span style="text-decoration: underline;">3' End index read primer (63 bp)</span></div><div>                                                                    <span style="font-weight: bold;">Index</span>                P7 Primer</div><div style="margin-bottom: 14.15pt;">GATCGGAAGAGCACACGTCTGAACTCCAGTCAC‐NNNNNN‐ATCTCGTATGCCGTCTTCTGCTTG</div><div style="margin-bottom: 14.15pt;">Adapters we use shown below in a .fa file (NEB_adaptors_set4.fa) will be used to trim sequences of adapters shown below:</div><div>&gt;5_end_adaptor_universal</div><div>AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT</div><div>&gt;3_end_index30</div><div style="margin-bottom: 14.15pt;">GATCGGAAGAGCACACGTCTGAACTCCAGTCACCACCGGATCTCGTATGCCGTCTTCTGCTTG</div><div style="margin-bottom: 14.15pt;">Using the package <a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf">trimmomatic</a>/0.36 to trim the reads</div><div># Run trimmomatic in single end mode ---&gt; trimmomatic SE</div><div style="margin-bottom: 14.15pt;">Trimmomatic SE [-threads &lt;threads&gt;]  &lt;input&gt; &lt;output&gt; &lt;step 1&gt; </div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>module load trimmomatic/0.36</div><div><span style="font-size: 9pt; color: rgb(51, 51, 51); font-family: Monaco;"><br/></span></div><div>trimmomatic SE -threads 4 control1_CACCGG.fastq.gz control1_trimmed.fastq.gz ILLUMINACLIP: NEB_adaptors_set4.fa:2:30:10 LEADING: 3 TRAILING:3 SLIDINGWINDOW:4:25 MINLEN:36</div></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">or with no adapters file specified</span></div><div style="margin-top: 1em; margin-bottom: 1em;"><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>trimmomatic SE -threads 4 control1_CACCGG.fastq.gz control1_trimmed.fastq.gz LEADING:3 TRAILING:3 SLIDINGWINDOW:4:25 MINLEN:36</div></div></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">After the trimming rerun FASTQC on the trimmed file to check the quality.</span></div><div style="margin-bottom: 0.0001pt;"><br/></div><div style="margin-bottom: 0.0001pt;"><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>module load fastqc/0.11.7</div><div>fastqc control1.trimmed.fastq.gz control2.trimmed.fastq.gz control3.trimmed.fastq.gz</div></div></div><div style="margin-bottom: 0.0001pt;"></div><h2>II. Align reads with Hisat2</h2><div><a href="https://ccb.jhu.edu/software/hisat2/manual.shtml">HISAT2</a>: similar algorithms as Bowtie2/Tophat2 but performs at much faster speeds</div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">First create index files for Hisat2 using the genome file. You only need to do this once. When you generate an index for a certain reference genome you can reuse the index for other alignments. </span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">Usage:</span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="font-size: 9.5pt; background-image: initial; background-position: initial; background-size: initial; background-repeat: initial; background-attachment: initial; background-origin: initial; background-clip: initial; color: rgb(102, 102, 102); font-family: &quot;Courier New&quot;; line-height: 107%;-en-paragraph:true;">hisat2-build [options]* &lt;reference_in&gt; &lt;ht2_base&gt;</span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">In command line:</span></div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>Module load hisat2/2.1.0</div><div><br/></div><div>hisat2-build /bigdata/zurniedenlab/dwill023/data/mouse/Mus_musculus.GRCm38.dna.primary_assembly.fa /bigdata/zurniedenlab/dwill023/data/mouse/hisat2_index/mm38</div></div><div style="margin-bottom: 14.15pt;"><br/></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">Run HISAT2. Usage:</span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="font-size: 9.5pt; background-image: initial; background-position: initial; background-size: initial; background-repeat: initial; background-attachment: initial; background-origin: initial; background-clip: initial; color: rgb(102, 102, 102); font-family: &quot;Courier New&quot;; line-height: 107%;-en-paragraph:true;">hisat2 [options]* -x &lt;hisat2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt; | --sra-acc &lt;SRA accession number&gt;} [-S &lt;hit&gt;]</span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="font-size: 9.5pt; background: white; color: rgb(102, 102, 102); font-family: &quot;Courier New&quot;; line-height: 107%;-en-paragraph:true;">-k &lt;int&gt; :</span> <span style="font-size: 9.5pt; background-image: initial; background-position: initial; background-size: initial; background-repeat: initial; background-attachment: initial; background-origin: initial; background-clip: initial; color: rgb(68, 68, 68); font-family: Verdana, sans-serif; line-height: 107%;-en-paragraph:true;">It searches for at most </span><span style="font-size: 9.5pt; background: white; color: rgb(68, 68, 68); line-height: 107%;-en-paragraph:true;">&lt;int&gt;</span><span style="font-size: 9.5pt; background: white; color: rgb(68, 68, 68); font-family: Verdana, sans-serif; line-height: 107%;-en-paragraph:true;"> distinct, primary alignments for each read.  Default: 5 (HFM) or 10 (HGFM).</span></div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>hisat2 -p 10 -x /bigdata/zurniedenlab/dwill023/data/mouse/hisat2_index/mm38  -U /bigdata/zurniedenlab/shared/mir690/rawdata/flowcell1145_lane1_pair1_CACCGG.fastq.gz -S /bigdata/zurniedenlab/dwill023/projects/rnaseq/results_mir690/control1 -k 1 --new-summary /rhome/dwill023/rnaseq/rnaseq/results/control1.txt</div></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">For multiple files use a shell script (hisat2_align.sh)</span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">Make the shell script executable by:</span></div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>$ Chmod +x hisat2_align.sh</div></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">Run shell script:</span></div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>$ bash hisat2_align.sh</div></div><div style="margin-top: 1em; margin-bottom: 1em;">Below is what is contained in the script:</div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;">#!/bin/bash</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;"> </span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;">#script to align RNA-seq of miR690 overexpression sequence files to the mouse genome.</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;"> </span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;"># Immediately stop on errors</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;"> </span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;">set -uex pipefail</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;"> </span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;"># The reference genome path</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;">REF=/bigdata/zurniedenlab/dwill023/data/mouse/hisat2_index/mm38</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext;"><br/></span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;">module load hisat2/2.1.0</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;"> </span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;">hisat2 -p 8 -k 1 -x $REF -U /bigdata/zurniedenlab/shared/mir690_rnaseq/rawdata/control1_trimmed.fastq.gz -S /bigdata/zurniedenlab/dwill023/projects/rnaseq/results_mir690/control1.sam --summary-file /rhome/dwill023/rnaseq/results/control.txt</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;"> </span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><span style="background: rgb(247, 247, 247); font-size: 9pt; letter-spacing: 0.15pt; border: 1pt none windowtext; color: rgb(51, 51, 51); font-family: Consolas; line-height: normal;">hisat2 -p 8 -k 1 -x $REF -U /bigdata/zurniedenlab/shared/mir690_rnaseq/rawdata/control2_trimmed.fastq.gz -S /bigdata/zurniedenlab/dwill023/projects/rnaseq/results_mir690/control2.sam --summary-file /rhome/dwill023/rnaseq/results/control2.txt</span><span style="font-size: 9pt; letter-spacing: 0.15pt; color: rgb(51, 51, 51); font-family: Consolas;"> </span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="font-weight: bold;">Run samtools to convert to bam then sort the bam file</span></div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>module load parallel</div><div>module load samtools</div><div><br/></div><div>ls *.sam | parallel --eta --verbose &quot;samtools view -@ 4 -Sb {} | samtools sort &gt; {.}.sorted.bam&quot;</div></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="font-weight: bold;">Alternatively:</span> Create a file (names.txt) containing all the names of the {}.fastq.gz files  </div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;"><span style="text-decoration: underline;">In command line</span> to run everything in parallel:</span></div><div style="margin-bottom: 0.0001pt; background: rgb(247, 247, 247);"><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>module load parallel</div><div>module load hisat2/2.1.0</div><div>module load samtools</div><div><br/></div><div>IDX=/bigdata/zurniedenlab/dwill023/data/mouse/hisat2_index/mm38</div><div><br/></div><div>cat names.txt | parallel &quot;hisat2 -p 10 -k 1 -x $IDX -U /bigdata/zurniedenlab/shared/mir690_rnaseq/rawdata/{}.fastq.gz -S /bigdata/zurniedenlab/dwill023/projects/rnaseq/results_mir690/{}.sam --summary-file /rhome/dwill023/rnaseq/results/{}.txt | samtools view -@ 4 -Sb /bigdata/zurniedenlab/dwill023/projects/rnaseq/results_mir690/{}.sam | samtools sort &gt; /bigdata/zurniedenlab/dwill023/projects/rnaseq/results_mir690/{}.sorted.bam&quot;</div></div></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;"> You don't need to index for the following steps. Only if you want to view the reads in</span> Integrative Genomics Viewer (IGV).</div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div># Run a samtools index on all BAM files</div><div>ls -1 /bigdata/zurniedenlab/dwill023/projects/rnaseq/results_mir690/*.bam | parallel samtools index {}</div></div><h1><span style="font-size: 14pt;">III. Generating counts</span></h1><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">This estimates the abundance of the reads that align over an interval. This is called read quantification and is required for gene expression analysis. The output is a count table that gives the number of reads within a gene feature. This is done with the tool</span> <span style="font-weight: bold;-en-paragraph:true;">featureCounts</span> <span style="-en-paragraph:true;">within the <a href="http://bioinf.wehi.edu.au/subread-package/SubreadUsersGuide.pdf">Subread package</a>.</span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">By default the featureCounts uses gene_id attributes from the GFF file.</span></div><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>module load subread/1.6.2</div><div>GTF=/bigdata/zurniedenlab/dwill023/data/mouse/*.gtf</div><div><br/></div><div># -T specifies the number of cores, -g to specify list generated with gene names instead of the default IDs.</div><div><br/></div><div>featureCounts -T 4 -a $GTF -g gene_name -o mir690_counts.txt /bigdata/zurniedenlab/dwill023/projects/rnaseq/results_mir690/*.bam</div><div><br/></div><div># Simplify the counts to have only gene names and counts</div><div>cat mir690_counts.txt | cut -f 1,6-20 &gt; 690simple_counts.txt</div></div><ol start="2"><li><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;"><span style="font-size: 10pt;">Getting</span> <span style="font-size: 10pt; font-weight: bold;">RPKM</span> <span style="font-size: 10pt;">from the raw counts. In R:</span></span></div></li></ol><div style="box-sizing: border-box; padding: 8px; font-family: Monaco, Menlo, Consolas, &quot;Courier New&quot;, monospace; font-size: 12px; color: rgb(51, 51, 51); border-radius: 4px; background-color: rgb(251, 250, 248); border: 1px solid rgba(0, 0, 0, 0.15);-en-codeblock:true;"><div>suppressPackageStartupMessages({</div><div>  library(readr)</div><div>  library(dplyr)</div><div>  library(ggplot2)</div><div>  library(DESeq2)</div><div>  library(edgeR)</div><div>})</div><div><br/></div><div># Getting the RPKM from the raw read counts</div><div>counts &lt;- read_delim(&quot;REST_simplecounts.txt&quot;, delim = &quot;\t&quot;) # load count matrices</div><div><br/></div><div>counts &lt;- as.data.frame(counts)</div><div># Remove duplicated rows in Geneid column</div><div>counts &lt;- counts %&gt;% distinct(Geneid, .keep_all = TRUE)</div><div><br/></div><div># remove the first column from counts</div><div>countdata &lt;- counts[,-(1)]</div><div><br/></div><div># Store Geneid as rownames</div><div>rownames(countdata) &lt;- counts[,1]</div><div>genes &lt;- countdata[,-(1)]</div><div>head(countdata)</div><div>colnames(genes)</div><div><br/></div><div># obtain RPKM</div><div>RPKM &lt;- rpkm(genes, gene.length = countdata$Length)</div><div><br/></div><div># Filter the RPKM for lowly expressed microRNAs.</div><div># which RPKMs are &gt; 0.1. A threshold that keeps genes with a raw count ~1</div><div>thresh &lt;- RPKM &gt; 0.1</div><div>head(thresh)</div><div><br/></div><div># Summary of how many TRUEs are in each row</div><div>table(rowSums(thresh))</div><div># we would like to keep genes that have at least 2 TRUES in each row of thresh</div><div>keep &lt;- rowSums(thresh) &gt;= 2</div><div><br/></div><div># Subset the rows of countdata to keep the more highly expressed genes</div><div>counts.keep &lt;- RPKM[keep,]</div><div>summary(keep)</div><div>dim(counts.keep)</div><div><br/></div><div># export the logcounts into a file</div><div>write.csv(counts.keep, &quot;RPKM_REST.csv&quot;)</div></div><div style="margin-top: 1em; margin-bottom: 1em;"></div><div style="margin-top: 1em; margin-bottom: 1em;"></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;"><span style="font-size: 14pt; font-weight: bold;">IV. DEG Analysis</span></span></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="font-size: 10pt;-en-paragraph:true;">Using R package Deseq2 following script in file </span><span style="font-size: 10pt;">DESeq2_code_mir690.R</span></div><ul><li><div style="margin-top: 1em; margin-bottom: 1em;"><span style="font-size: 10pt;">PCA plot (Principal component analysis): A form of dimension reduction that converts the correlations (or lack of) among all samples an plots them onto a 2D plot. Samples that are highly correlated cluster together. Differences along the first axis (PC1) are more important than differences along the PC2 axis. </span><span style="font-size: 10pt;">A measure of the variance between samples. </span><span style="font-size: 10pt;">Heatmaps generated using the rlog transformation for samples &lt; 30. For samples &gt; 30 use the VST normalization. </span></div></li><li><div><span style="font-size: 10pt;">MA-plot: </span> <span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); font-size: 10pt; color: rgb(34, 34, 34); font-variant-caps: normal; font-variant-ligatures: normal;"> </span><span style="letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; font-size: 10pt; color: rgb(34, 34, 34); font-variant-caps: normal; font-variant-ligatures: normal;">plot visualizes the gene expression differences between two samples. </span></div></li><li><div><font color="#222222"><span style="font-size: 10pt;">Volcano plot: Generates visualization</span> <span style="font-size: 13.3333px;">similar</span><span style="font-size: 10pt;"> to MA plot but with more details (pvalues, gene names.)</span></font></div></li><li><div><span style="font-size: 10pt; color: rgb(34, 34, 34);">Gene Set Enrichment Analysis with ClusterProfiler</span></div></li></ul><div><br/></div></div><div>For the results if you see p-value &amp; p.adjust values of NA it indicates outliers detected by Cook’s distance. NA only for p.adjust means the gene is filtered by automatic independent filtering for having a low mean normalized count. </div></span>
</div></body></html> 